components/core/Profile.js

```diff
@@ -103,78 +101,495 @@ const STYLES_DESCRIPTION = css`
-  margin: 160px auto 64px auto;
+  margin: 64px auto 64px auto;
   height: 1px;
   width: 80px;
   background-color: ${Constants.system.gray};
 `;
 
+const STYLES_BUTTON = css`
+  margin-bottom: 32px;
+
+  @media (max-width: ${Constants.sizes.mobile}px) {
+    margin-bottom: 16px;
+  }
+`;
+
+const STYLES_ITEM_BOX = css`
+  position: relative;
+  justify-self: end;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  padding: 8px;
+  margin-right: 16px;
+  color: ${Constants.system.darkGray};
+
+  @media (max-width: ${Constants.sizes.mobile}px) {
+    margin-right: 8px;
+  }
+`;
+
+const STYLES_USER_ENTRY = css`
+  display: grid;
+  grid-template-columns: auto 1fr;
+  align-items: center;
+  font-size: ${Constants.typescale.lvl1};
+  cursor: pointer;
+  ${"" /* border: 1px solid ${Constants.system.lightBorder}; */}
+  border-radius: 4px;
+  margin-bottom: 8px;
+  background-color: ${Constants.system.white};
+`;
+
+const STYLES_USER = css`
+  display: grid;
+  grid-template-columns: auto 1fr;
+  align-items: center;
+  margin: 16px;
+  color: ${Constants.system.brand};
+  font-family: ${Constants.font.medium};
+  font-size: ${Constants.typescale.lvl1};
+
+  @media (max-width: ${Constants.sizes.mobile}px) {
+    margin: 12px 16px;
+  }
+`;
+
+const STYLES_DIRECTORY_PROFILE_IMAGE = css`
+  background-color: ${Constants.system.foreground};
+  background-size: cover;
+  background-position: 50% 50%;
+  height: 24px;
+  width: 24px;
+  margin-right: 16px;
+  border-radius: 4px;
+`;
+
+const STYLES_MESSAGE = css`
+  color: ${Constants.system.black};
+  overflow: hidden;
+  white-space: nowrap;
+  text-overflow: ellipsis;
+
+  @media (max-width: 1000px) {
+    display: none;
+  }
+`;
+
+const STYLES_DIRECTORY_NAME = css`
+  overflow: hidden;
+  white-space: nowrap;
+  text-overflow: ellipsis;
+`;
+
+const STYLES_COPY_INPUT = css`
+  pointer-events: none;
+  position: absolute;
+  opacity: 0;
+`;
+
+function UserEntry({ user, button, onClick, message, external, url }) {
+  return (
+    <div key={user.username} css={STYLES_USER_ENTRY}>
+      {external ? (
+        <a css={STYLES_USER} style={{ textDecoration: "none" }} href={url}>
+          <div
+            css={STYLES_DIRECTORY_PROFILE_IMAGE}
+            style={{ backgroundImage: `url(${user.data.photo})` }}
+          />
+          <span css={STYLES_DIRECTORY_NAME}>
+            {user.data.name || `@${user.username}`}
+            {message ? <span css={STYLES_MESSAGE}>{message}</span> : null}
+          </span>
+        </a>
+      ) : (
+        <div css={STYLES_USER} onClick={onClick}>
+          <div
+            css={STYLES_DIRECTORY_PROFILE_IMAGE}
+            style={{ backgroundImage: `url(${user.data.photo})` }}
+          />
+          <span css={STYLES_DIRECTORY_NAME}>
+            {user.data.name || `@${user.username}`}
+            {message ? <span css={STYLES_MESSAGE}>{message}</span> : null}
+          </span>
+        </div>
+      )}
+      {button}
```

Comment:
This menu should be hidden on external, so "{external ? null : button}"

Reply:
why should we hide the menu on external?

Reply:
Because the menu has two things in it: "follow/unfollow" and "copyURL". Copy url is unnecessary and I'll take it out of all menus since we now have nice urls that you can copy directly from the url bar. Follow /unfollow is not alllowed for external